lib/resty/openssl/x509/csr.lua

```diff
@@ -190,7 +190,8 @@ end
 
 local function modify_extension(replace, ctx, nid, toset, crit)
   local extensions_ptr = stack_ptr_type()
-  extensions_ptr[0] = C.X509_REQ_get_extensions(ctx)
+  local extension = C.X509_REQ_get_extensions(ctx)
+  extensions_ptr[0] = extension
```

Comment:
trying to understand this change, is this more for readability, or a bugfix?

Reply:
This is a bug.
extensions_ptr[0] = C.X509_REQ_get_extensions(ctx)
the extensions_ptr[0]  is a cdata, it does not hold a reference to the object create by C.X509_REQ_get_extensions(ctx).
So the object created by C.X509_REQ_get_extensions(ctx) maybe GC collected in any time.

Reply:
I see the point, it would be the case that if the object is created by luajit, for example, a string or a ffi.new(). But in this case, `C.X509_REQ_get_extensions` allocates memory in C side, luajit should not know how to free it right?

Reply:
I now understand.